Tests/NooraTests/Components/YesOrNoChoicePromptTests.swift

```diff
@@ -0,0 +1,70 @@
+import Mockable
+import Testing
+
+@testable import Noora
+
+struct YesOrNoChoicePromptTests {
+    var subject: YesOrNoChoicePrompt!
+    var keyStrokeListener = MockKeyStrokeListening()
+    var renderer = MockRendering()
+    var terminal = MockTerminaling()
+
+    @Test func renders_the_right_content() throws {
+        // Given
+        let subject = YesOrNoChoicePrompt(
+            title: "Authentication",
+            question: "Would you like to authenticate?",
+            defaultAnswer: true,
+            collapseOnSelection: true,
+            theme: NooraTheme.test(),
+            terminal: terminal,
+            renderer: renderer,
+            standardPipelines: StandardPipelines(),
+            keyStrokeListener: keyStrokeListener
+        )
+
+        var inRawMode: (() throws -> Void)!
+        given(terminal).inRawMode(.any).willReturn()
+        when(terminal).inRawMode(.matching { callback in
+            inRawMode = callback
+            return true
+        }).perform {
+            try? inRawMode()
+        }
```

Comment:
Can't this be simplified to
```suggestion
        given(terminal).inRawMode(.any).willProduce({ $0() })
```

Same goes for `onKeyPress` and the other test suite